tests/e2etest-xsqlinstance/main.k

```diff
@@ -0,0 +1,111 @@
+import models.io.upbound.aws.v1beta1 as awsv1beta1
+import models.io.upbound.dev.meta.v1alpha1 as metav1alpha1
+import models.io.upbound.platform.aws.v1alpha1 as platformawsv1alpha1
+
+_items = [
+    metav1alpha1.E2ETest{
+        metadata.name: "xsqlinstances"
+        spec= {
+            crossplane.autoUpgrade.channel: "Rapid"
+            defaultConditions: ["Ready"]
+            manifests: [
+                platformawsv1alpha1.XSQLInstance{
+                    metadata = {
+                        name = "configuration-aws-database-mariadb"
+                    }
+                    spec = {
+                        parameters = {
+                            region = "us-west-2"
+                            engine = "mariadb"
+                            engineVersion = "10.11.6"
+                            storageGB = 5
+                            autoGeneratePassword = True
+                            passwordSecretRef = {
+                                namespace = "default"
+                                name = "mariadbsecret"
+                                key = "password"
+                            }
+                            networkRef = {
+                                id = "configuration-aws-database"
+                            }
+                        }
+                        writeConnectionSecretToRef = {
+                            name = "configuration-aws-database-mariadb"
+                            namespace = "default"
+                        }
+                    }
+                }
+                platformawsv1alpha1.XSQLInstance{
+                    metadata = {
+                        name = "configuration-aws-database-postgres"
+                    }
+                    spec = {
+                        parameters = {
+                            region = "us-west-2"
+                            engine = "postgres"
+                            engineVersion = "13.18"
+                            storageGB = 5
```

Comment:
```suggestion
                            storageGB = 5
                            autoGeneratePassword = True
```

Reply:
I'm confused, i'm still under the impression that we first migrate as is and do the nitpicking after? 🤔

Reply:
It simplifies the (outdated) example, and you can remove secrets from extra resources. 

So yes, we migrate the composition logic as-is. The examples can be simplified right away.

Reply:
ok, makes sense